java.source.base/src/org/netbeans/modules/java/source/indexing/JavaCustomIndexer.java

```diff
@@ -130,6 +130,7 @@
     private static final String APT_SOURCE_OUTPUT = "apSrcOut"; //NOI18N
     private static final Pattern ANONYMOUS = Pattern.compile("\\$[0-9]"); //NOI18N
     private static final ClassPath EMPTY = ClassPathSupport.createClassPath(new URL[0]);
+    private static final String PROP_NB_JAVAC = "nb-javac.installed"; //NOI18N
```

Comment:
Wild, terrible hack.

Wouldn't it be more constructive to record version of `javac` that is used for indexing and if the version changes (due to `nbjavac` being installed or due to change of `JDK` etc.) reparse?

Reply:
Regarding javac version, good idea actually. Tried to implement.

For Maven, I was going to say it will work (as the patch modified core of the CoS implementation), but then it turned out Maven is bypassing that. So trying a different approach - adding API to check is CoS is supported, and disabling the feature if it is not.